terraform/stacks/unimelb/data_archive/analysis_archive.tf

```diff
@@ -116,6 +116,23 @@ data "aws_iam_policy_document" "analysis_archive" {
     ])
   }
 
+  # Allow the presign user to presign URLs from this bucket.
+  statement {
+    sid = "orcabus_file_manager_presign_access"
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${local.account_id_prod}:user/${local.orcabus_file_manager_presign_user}"]
+    }
+    actions = sort([
+      "s3:ListBucket",
+      "s3:GetObject"
+    ])
+    resources = sort([
+      aws_s3_bucket.analysis_archive.arn,
+      "${aws_s3_bucket.analysis_archive.arn}/*",
+    ])
+  }
+
```

Comment:
It might not be the intent to allow presigning from the analysis archive bucket. I can remove this if so. Same goes for the fastq archive bucket.

Reply:
Presigning would only work for classes that are instantly retrieval anyway, e.g. `Glacier IR`.

Reply:
Ideally sharing would only happen from the BYOB, but until we have sorted and automated that, there will be sharing requests for data in archive. 
However, we agreed restore FASTQs into BYOB if/when that's needed, so there is no need to include the fastq archive. (short lived purls are already possible for data that's not in archive tiers, right?)

Reply:
> However, we agreed restore FASTQs into BYOB if/when that's needed, so there is no need to include the fastq archive. (short lived purls are already possible for data that's not in archive tiers, right?)

There's no way to signal to the API which credentials to sign with - as of now, for https://github.com/umccr/orcabus/pull/862, it's always using the user access key.

I could do something like: try access key credentials -> check if URL works, if not, try role credentials. I was thinking of adding something like this in anyway, because I don't want the filemanager to accidentally return unusable presigned URLs

Reply:
> There's no way to signal to the API which credentials to sign with - as of now, for https://github.com/umccr/orcabus/pull/862, it's always using the user access key.

This now works with https://github.com/umccr/orcabus/pull/862/commits/8301a6d38e8101b64002a9fbff438fda547f79ca